### PR TITLE
feat(Multi project runs): Multi project runs using solution file

### LINKED ISSUE #459: Multi project runs: Mutate more projects at once
In a situation where you use both normal unittests and SpecFlow tests, you can end up with two test projects, testing the same implementation project.

Now you have to run stryker twice and you end up with two results. Which probably have a certain amount of overlap.

It would be helpful if you can merge the results, or even better, run both projects at once, so you have only one result.

### LINKED ISSUE #1076: DefineConstants are not parsed correctly
Stryker throws a compile error during the initial test run on vstest.
No special configurations, just specified the path for my *.csproj file on cmd. 
Attached the file with logs on the cmd during test run.
[StrykerLogs.txt](https://github.com/stryker-mutator/stryker/files/4628241/StrykerLogs.txt)
 
Please let me know if you need additional information.

### PR DESCRIPTION
**Pull request by richardwerkman**
blocked by #1136

closes #459 
closes #1076

Also utilizes the new interfaces from buildalyzer to make testing possible for the parts that use buildalyzer. I'll wait to add more tests after merging this PR as this one is already quite large.

Needs some performance tweaks:
- The initial build is called for each test project, it could run for the whole solution once
- Initial testing could also be done in one run??
- Buildalyzer analyses the projects twice now. Once in the projectOrchestrator and once in the InputFileResolver. It would be better to pass the buildalyzer results to reuse. 

Open questions:
 - Should the thresholds count for the whole solution or for each project seperate? For example:
    - If one project scores 50% and the other 100% with equal mutants, would the run succeed with threshold break: 60?

### CODE REVIEW

File: src/Stryker.Core/Stryker.Core/Reporters/JsonReporter/JsonReportFileComponent.cs [hunk 1]
```diff
@@ -22,7 +22,7 @@ private JsonReportFileComponent(string language, string source, ISet<JsonMutant>
 
         public JsonReportFileComponent(FileLeaf file, ILogger logger = null)
         {
-            var log = logger ?? ApplicationLogging.LoggerFactory.CreateLogger<JsonReportFileComponent>();
+            logger = logger ?? ApplicationLogging.LoggerFactory.CreateLogger<JsonReportFileComponent>();
```
Comment (Mobrockers) [inline src/Stryker.Core/Stryker.Core/Reporters/JsonReporter/JsonReportFileComponent.cs:25]: Did sonarqube not mention that you should not overwrite a method parameter value?

Comment (richardwerkman) [reply src/Stryker.Core/Stryker.Core/Reporters/JsonReporter/JsonReportFileComponent.cs:25]: No

Comment (Mobrockers) [reply src/Stryker.Core/Stryker.Core/Reporters/JsonReporter/JsonReportFileComponent.cs:25]: Mhmm 🤔

**Pull request merged**